artcommon/artcommonlib/util.py

```diff
@@ -151,6 +151,17 @@ def isolate_rhel_major_from_version(version: str) -> Optional[int]:
     return None
 
 
+def isolate_rhel_major_from_distgit_branch(branch: str) -> Optional[int]:
+    """
+    E.g. 'rhaos-4.16-rhel-9' => 9
+    """
+
+    match = re.fullmatch(r"^rhaos-(\d+).(\d+)-rhel-(\d+)", branch)
```

Comment:
Please escape the `.`. Maybe also do not put capture groups around major and minor. as this match is irrelevant. 

`"^rhaos-\d+\.\d+-rhel-(\d+)`